extensions/mongo.md

```diff
@@ -41,14 +45,55 @@ public class AxonConfig {
   
     // The MongoEventStorageEngine stores each event in a separate MongoDB document.
     @Bean
-    public EventStorageEngine storageEngine(MongoClient client) {
+    public EventStorageEngine storageEngine(MongoDatabaseFactory factory,
+                                            TransactionManager transactionManager) {
         return MongoEventStorageEngine.builder()
-                                      .mongoTemplate(DefaultMongoTemplate.builder()
-                                                                         .mongoDatabase(client)
-                                                                         .build())
+                                      .mongoTemplate(SpringMongoTemplate.builder()
+                                                                        .factory(factory)
+                                                                        .build())
+                                      .transactionManager(transactionManager)
                                       // ...
                                       .build();
     }
 }
 ```
 
+## Configuration in Spring Boot
+
+This extension can be added as a Spring Boot starter dependency to your project using group id `org.axonframework.extensions.mongo` and artifact id `axon-mongo-spring-boot-starter`. When using the autoconfiguration, by  default the following components will be created for you automatically:
+* A `MongoTransactionManager` to enable transactions with Mongo.
+* A `SpringMongoTransactionManager`, this is the wrapped Spring mongo transaction manager, and will also be injected where applicable in other components created by the auto-config.
+* A `SpringMongoTemplate`, this will use a `MongoDatabaseFactory` that should be available. To use transaction with Mongo the collections need to be accessed in a certain way, and this component makes sure of that.
+* A `MongoTokenStore`, this will be used by the event processors to can keep track which events have been processed, and which segments are claimed.
+* A `MongoSagaStore`, this will be used to store and retrieve saga's.
+
+It's also possible to autoconfigure the `StorageStrategy` and `EventStorageEngine` by setting the `mongo.event-store.enabled` to true. The creation of the token store and the saga store can be turned off by setting `mongo.token-store.enabled` or `mongo.saga-store.enabled` to false. It's also possible to use a different database for the axon collections than the default the `MongoDatabaseFactory` uses by setting the `axon.mongo.database-name` property.
```

Comment:
What may be a helpful addition is a sample, or several samples, of what the properties file may look like when enabling/disabling some of the beans.